Makefile

```diff
@@ -109,10 +110,18 @@ all: $(LIBRARY_RELEASE_$(OS)) libkrun.pc
 
 debug: $(LIBRARY_DEBUG_$(OS)) libkrun.pc
 
+ifeq ($(SYSROOT_LINUX),)
+# Build on Linux host
+CC_LINUX=$(CC)
+else
+# Cross-compile on macOS with the LLVM linker (brew install lld)
+CC_LINUX=clang -target $(ARCH)-linux-gnu -fuse-ld=lld -Wl,-strip-debug --sysroot $(SYSROOT_LINUX) -Wno-c23-extensions
```

Comment:
Please use `/usr/bin/clang` instead of `clang` here. Homebrew puts a wrapper in PATH that messes up with the command line, so we need to avoid by using an absolute path.